### PR TITLE
Move btc-server into macbeth as a crate

### PR DESCRIPTION
**Pull request by scottmillner**

### CODE REVIEW

File: bin/btc-server/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+[package]
+name = "btc-server"
+version = "0.1.0"
+edition = "2021"
+
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
+
+[dependencies]
+reth-btc-wallet = { path = "../../crates/btc-wallet" }
```
Comment (arminsabouri) [inline bin/btc-server/Cargo.toml]: Referencing the work space would be prefered over a path here

File: bin/btc-server/key.hex [hunk 2]
```diff
@@ -0,0 +1 @@
+e9e3969ceeb49c13f018acba996c79af35f920d2520bd2a36df03208936cc717
```
Comment (arminsabouri) [inline bin/btc-server/key.hex]: Lets remove this file. We should only need key.template.hex and the instructions have the developer copy it over to key.hex

**Review by arminsabouri — APPROVED** [review]
This is awesome! Just had a couple review items

File: bin/btc-server/Cargo.toml (hunk 1 above)
Comment (scottmillner) [reply bin/btc-server/Cargo.toml]: made change

File: bin/btc-server/key.hex (hunk 2 above)
Comment (scottmillner) [reply bin/btc-server/key.hex]: made change

**Pull request merged**